### PR TITLE
Keyed mapUsingContext operations should take key as one of the parameters

### PR DESCRIPTION
**Pull request by cangencer**

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/TriPredicate.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package com.hazelcast.jet.function;
+
+import java.util.function.Predicate;
+
+/**
+ * Represents a predicate which accepts three arguments. This
+ * is the three-arity specialization of {@link Predicate}.
+ **/
+@FunctionalInterface
+public interface TriPredicate<T, U, V> {
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/TriPredicate.java]: `TriPredicate` is now lacking the composition methods.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * {@code Serializable} variant of {@link TriPredicate}.
+ */
+public interface DistributedTriPredicate<T, U, V> extends TriPredicate<T, U, V>, Serializable {
+
+    /**
+     * Returns a composite predicate which evaluates the
+     * equivalent of {@code this.test(t, u, v) && other.test(t, u, v)}.
+     */
+    default DistributedTriPredicate<T, U, V> and(DistributedTriPredicate<? super T, ? super U, ? super V> other) {
+        checkNotNull(other, "other");
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java]: It should also `checkSerializable(other)`

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * {@code Serializable} variant of {@link TriPredicate}.
+ */
+public interface DistributedTriPredicate<T, U, V> extends TriPredicate<T, U, V>, Serializable {
+
+    /**
+     * Returns a composite predicate which evaluates the
+     * equivalent of {@code this.test(t, u, v) && other.test(t, u, v)}.
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java]: Add "The `other` predicate must not be null".

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java (hunk 2 above)
Comment (viliam-durina) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java]: I don't think so, the method taking the resulting predicate should.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java (hunk 3 above)
Comment (cangencer) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java]: better to add `@Nonnull` annotations I think

Comment (mtopolnik) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedTriPredicate.java]: Indeed :)

**Pull request merged**